src/Forms/GridField/GridFieldLazyLoader.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace SilverStripe\Forms\GridField;
+
+use SilverStripe\Forms\FormField;
+use SilverStripe\Forms\TabSet;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\ORM\Filterable;
+use SilverStripe\ORM\Limitable;
+use SilverStripe\ORM\SS_List;
+
+/**
+ * GridFieldLazyLoader alters the {@link GridField} behavior to delay rendering of rows until the tab containing the
+ * GridField is selected by the user.
+ *
+ * @see GridField
+ */
+class GridFieldLazyLoader implements GridField_DataManipulator, GridField_HTMLProvider
+{
+
+    /**
+     * @inheritDoc
+     */
+    public function getManipulatedData(GridField $gridField, SS_List $dataList)
+    {
+        // If we are lazy loading an empty the list
+        if ($this->isLazy($gridField)) {
+            if ($dataList instanceof Filterable) {
+                // If our original list can be filtered, filter out all results.
+                $dataList = $dataList->byIDs([-1]);
+            } else {
+                // If not, create an empty list instead.
+                $dataList = ArrayList::create([]);
+            }
+        }
+        return $dataList;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getHTMLFragments($gridField)
+    {
+        $gridField->addExtraClass($this->isLazy($gridField) ?
+            'grid-field-lazy-loadable' :
+            'grid-field-lazy-loaded');
```

Comment:
Should these be BEM convention since `loadable/loaded` seem like modifiers? `grid-field--lazy-loadable` `grid-field--lazy-loaded`

Reply:
Done. Matching changes have been made to `silverstripe/admin` as well.